### DISCUSSION TITLE
When pruning a pruned model,It would be better if not ctreate a new wrapper

### DISCUSSION THREAD

**Discussion by trotsky1997** [microsoft/nni#4216] [Q&A] [answered]
**at beginning:**
` Linear(in_features=128, out_features=10, bias=True)`
after pruned once:
```
 (2): PrunerModuleWrapper(
        (module): Linear(in_features=128, out_features=10, bias=True)
```

 **it works.butafter pruned twice:**
```
 (2): PrunerModuleWrapper(
      (module): PrunerModuleWrapper(
        (module): Linear(in_features=128, out_features=10, bias=True)
      )
```
**it would stucked at apply mask to weight, bias**
```
~\AppData\Roaming\Python\Python39\site-packages\nni\compression\pytorch\compressor.py in forward(self, *inputs)
    311     def forward(self, *inputs):
    312         # apply mask to weight, bias
--> 313         self.module.weight.data = self.module.weight.data.mul_(self.weight_mask)
    314         if hasattr(self.module, 'bias') and self.module.bias is not None:
    315             self.module.bias.data = self.module.bias.data.mul_(self.bias_mask)
```

solvement?
1. set PrunerModuleWrapper.weight = PrunerModuleWrapper.module.weight
or
2. take the wrapped modules out from that internel wrappers.
3. ...

**Comment by Davidxswang** [comment, marked as answer]
Maybe you can call `_unwrap_model` method (defined in the class `Compressor`) before pruning the same model again. The `_unwrap_model` method will take the module out of the wrapper for each wrapped module in the model being pruned. @trotsky1997

**Comment by trotsky1997** [comment]
> Maybe you can call `_unwrap_model` method (defined in the class `Compressor`) before pruning the same model again. The `_unwrap_model` method will take the module out of the wrapper for each wrapped module in the model being pruned. @trotsky1997

yes,i have use this method already,that works fine
